command/detect.go

```diff
@@ -36,7 +36,7 @@ func DetectCommand(detect libjavabuildpack.Detect, metadata riff_buildpack.Metad
 		return false, err
 	}
 	info, err := os.Stat(path)
-	if err != nil {
+	if err != nil || info.IsDir() {
```

Comment:
This is sort of the anti-test.  Would it be more accurate to look to ensure that it's a regular file and possibly a symlink?  If you look at the modes supported the `info` could potentially be non-directory but be character devices, named pipes, devices, sockets, etc.  The test isn't robust, but you might value this simplicity over the strictest test.  Something to think about.

Reply:
You're absolutely right. I threw that off quickly because I hit the case of a directory for an npm module. Will change